src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -454,18 +459,18 @@ private static boolean skipDoubleBraceInstInit(Details details) {
      * @return true if block has single-line format and is alone on a line.
      */
     private static boolean isBlockAloneOnSingleLine(Details details) {
-        final DetailAST rcurly = details.rcurly;
-        final DetailAST lcurly = details.lcurly;
         DetailAST nextToken = details.nextToken;
-        while (nextToken.getType() == TokenTypes.LITERAL_ELSE) {
+        while (nextToken != null && nextToken.getType() == TokenTypes.LITERAL_ELSE) {
```

Comment:
Can the next token ever become `null` or is it always the first instance that is `null`?
Basically asking if the code should be switched to:
````
if (nextToken != null) {
    while (...) {
        ...
    }
}
````

Reply:
Yeah, it can be null only at first instance since, after `if else` there would be atleast  another `rcurly` for end of class or interface. Thanks for pointing out.